ExampleMod/Content/Projectiles/ExampleSpearProjectile.cs

```diff
@@ -0,0 +1,94 @@
+﻿using ExampleMod.Content.Dusts;
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ModLoader;
+using static Terraria.ModLoader.ModContent;
+
+namespace ExampleMod.Content.Projectiles
+{
+	public class ExampleSpearProjectile : ModProjectile
+	{
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Spear");
+		}
+
+		public override void SetDefaults() {
+			projectile.width = 18; // The width of the projectile's hitbox in pixels.
+			projectile.height = 18; // The height of the projectile's hitbox in pixels.
+			projectile.aiStyle = 19; // Selects which vanilla code to use for the AI method. Modders can use vanilla aiStyle to mimic AI code already in the game.
+			projectile.penetrate = -1; // The Peneration of the projectile. -1 is infinite peneration.
+			projectile.scale = 1.3f; // The size multiplier of the projectile's sprite while the item is being used.
+			projectile.alpha = 0; // How transparent to draw this projectile. 0 to 255. 255 is completely transparent.
+
+			projectile.hide = true; 
+			projectile.ownerHitCheck = true;
+			projectile.melee = true; // Determines which crit chance will influence the damage of this projectile
+			projectile.tileCollide = false; // The projectile will collide with tiles.
+			projectile.friendly = true; // If True, this projectile will hurt enemies.
+		}
+
+		// In here the AI uses this example, to make the code more organized and readable
+		public float movementFactor // Change this value to alter how fast the spear moves
+		{
+			get => projectile.ai[0];
+			set => projectile.ai[0] = value;
+		}
+
+		// It appears that for this AI, only the ai0 field is used!
+		public override void AI() {
+			// Since we access the owner player instance so much, it's useful to create a helper local variable for this
+			// Sadly, Projectile/ModProjectile does not have its own
+			Player projOwner = Main.player[projectile.owner];
+			// Here we set some of the projectile's owner properties, such as held item and itemtime, along with projectile direction and position based on the player
+			Vector2 ownerMountedCenter = projOwner.RotatedRelativePoint(projOwner.MountedCenter, true);
+			projectile.direction = projOwner.direction;
+			projOwner.heldProj = projectile.whoAmI;
+			projOwner.itemTime = projOwner.itemAnimation;
```

Comment:
what does this do?

Reply:
if you're talking about itemtime = itemanimation it basically Sets item time to the frames of the item animation while used.